slackclient/_server.py

```diff
@@ -50,6 +50,7 @@ def rtm_connect(self, reconnect=False):
                 if not reconnect:
                     self.parse_slack_login_data(login_data)
                 self.connect_slack_websocket(self.ws_url)
+                self.connected = True
```

Comment:
self.connected should definitely represent reality, but some further changes are necessary, such as detection of a disconnected state.